lib/manager/poetry/artifacts.ts

```diff
@@ -37,6 +39,50 @@ function getPythonConstraint(
   return undefined;
 }
 
+function getPoetrySources(content: string, fileName: string): PoetrySource[] {
+  let pyprojectFile: PoetryFile;
+  try {
+    pyprojectFile = parse(content);
+  } catch (err) {
+    logger.debug({ err }, 'Error parsing pyproject.toml file');
+    return [];
+  }
+  if (!pyprojectFile.tool?.poetry) {
+    logger.debug(`{$fileName} contains no poetry section`);
+    return [];
+  }
+
+  const sources = pyprojectFile.tool?.poetry?.source;
+  const sourceArray: PoetrySource[] = [];
+  for (const source of sources) {
+    if (source.name && source.url) {
+      sourceArray.push({ name: source.name, url: source.url });
+    }
+  }
+  return sourceArray;
+}
+
+function getSourceCredentialVars(
+  pyprojectContent: string,
+  packageFileName: string
+): { [s: string]: string } {
+  const poetrySources = getPoetrySources(pyprojectContent, packageFileName);
+  const envVars = {};
```

Comment:
```suggestion
  const envVars: { [s: string]: string } = {};
```

Reply:
Thanks for this suggestion; haven't used typescript before so not sure what kind of syntax is required 🙂